### PR TITLE
Fix #2003 Deleting an event in 'editevent' without checking checkbox …

### LINKED ISSUE #2003: Deleting an event in 'editevent' without checking checkbox throws errors
When trying to delete an event from the edit event form without checking the checkbox, multiple errors are thrown:

    The name for the event is missing. Please enter an event name.
    The description for the event is missing. Please enter an event description.

All data fields (name, description, time) are also cleared.

Original thread: http://community.mybb.com/thread-168437.html

### PR DESCRIPTION
**Pull request by psde**
…throws errors

Adds a new action `do_deleteevent` for deleting events and displays a proper error message (with a new language string) when the checkbox is not checked. The JavaScript function `MyBB.deleteEvent(eid)` is also using this action now. The action `do_editevent` is not capable of deleting events anymore.

https://github.com/mybb/mybb/issues/2003

### CODE REVIEW

**Comment by psde** [comment]
I am looking for some guidance and suggestions on how to improve the current state of `calendar.php`. Currently every action is using (almost) the same code for checking permissions.

Also, I am worried about breaking compatibility with this pull request, as `do_editevent` is not capable of deleting events anymore - which is maybe used by a plugin?

**Comment by Destroy666x** [comment]
> I am looking for some guidance and suggestions on how to improve the current state of calendar.php. Currently every action is using (almost) the same code for checking permissions.

Feel free to discuss it on our forums in 1.8 suggestions & feedback category.

> Also, I am worried about breaking compatibility with this pull request, as do_editevent is not capable of deleting events anymore - which is maybe used by a plugin?

I'm quite sure there aren't any plugins it breaks (public at least) and if they are, they can be fixed easily. And I don't think there is another fix which won't break anything, but feel free to share if you have something else on your mind.

However, there are still 2 corrections that need to be made:
1) Bump the version of the modified template to `1806`
2) Bump the version of the `general.js` file in all templates (and their versions too) and all ACP files

**Comment by JN-Jones** [comment]
@psde?

**Comment by psde** [comment]
Sorry for the late update. I bumped the versions on the theme and general.js. Which ACP files do I have to modify?

File: install/resources/mybb_theme.xml [hunk 1]
```diff
@@ -13871,7 +13871,7 @@ $(function() {
 <meta http-equiv="Content-Script-Type" content="text/javascript" />
 <script type="text/javascript" src="{$mybb->asset_url}/jscripts/jquery.js?ver=1804"></script>
 <script type="text/javascript" src="{$mybb->asset_url}/jscripts/jquery.plugins.min.js?ver=1804"></script>
-<script type="text/javascript" src="{$mybb->asset_url}/jscripts/general.js?ver=1804"></script>
+<script type="text/javascript" src="{$mybb->asset_url}/jscripts/general.js?ver=1805"></script>
```
Comment (Destroy666x) [inline install/resources/mybb_theme.xml]: Should be `1.8.6`

**Comment by Destroy666x** [comment]
https://github.com/mybb/mybb/blob/feature/admin/inc/class_page.php#L111
https://github.com/mybb/mybb/blob/feature/admin/inc/class_page.php#L389
https://github.com/mybb/mybb/blob/feature/admin/inc/class_page.php#L521

Could be done here too:
https://github.com/mybb/mybb/blob/feature/install/resources/output.php#L43

**Comment by psde** [comment]
Thanks, I see what you meant. I updated the PR to reflect the suggested changes.

File: calendar.php [hunk 2]
```diff
@@ -596,6 +596,46 @@
 		// Redirect back to the main calendar view.
 		redirect("calendar.php", $lang->redirect_eventdeleted);
 	}
+	else
+	{
+		error($lang->delete_no_checkbox);
```
Comment (Destroy666x) [inline calendar.php]: `delete_no_checkbox` doesn't seem to be defined anywhere, at least I can't find it in my local installation with this PR.

File: install/resources/mybb_theme.xml [hunk 3]
```diff
@@ -13871,7 +13871,7 @@ $(function() {
 <meta http-equiv="Content-Script-Type" content="text/javascript" />
 <script type="text/javascript" src="{$mybb->asset_url}/jscripts/jquery.js?ver=1804"></script>
 <script type="text/javascript" src="{$mybb->asset_url}/jscripts/jquery.plugins.min.js?ver=1804"></script>
-<script type="text/javascript" src="{$mybb->asset_url}/jscripts/general.js?ver=1804"></script>
+<script type="text/javascript" src="{$mybb->asset_url}/jscripts/general.js?ver=1806"></script>
```
Comment (Destroy666x) [inline install/resources/mybb_theme.xml]: And the last thing left (if noone notices anything else) - the template version of this modified template - `headerinclude` should be bumped too.

**Comment by Destroy666x** [comment]
Oh, never mind, one more thing - I guess we should add at least 1 hook to the new action.

**Comment by JN-Jones** [comment]
@psde

File: calendar.php (hunk 2 above)
Comment (psde) [inline calendar.php]: I added it to the language file: https://github.com/mybb/mybb/pull/2006/files#diff-505deb1daf1b38b7bace47d542d761e5R93 - is there anything else I need to do?

Comment (JN-Jones) [inline calendar.php]: The `headerinclude` template @Destroy666x mentioned below and the new plugin hook.

**Comment by psde** [comment]
All versions should be bumped now, and I also added the requested hooks. The question about the language file is still unanswered. Do I have to bump a version for the modified language file as well?

Should I open an issue on the documentation and submit a PR for the new hooks introduced?

**Comment by JN-Jones** [comment]
> Do I have to bump a version for the modified language file as well?

No, language files are handled by us before creating a relase.

> Should I open an issue on the documentation ~~and submit a PR for the new hooks introduced~~? Seems like the hooks are generated automatically.

They were generated automatically but they're not generated "on the fly". Feel free to generate an issue, however instead of adding the new hooks manually the Hook Finder should be rerun before 1.8.6 gets released.

**Comment by Destroy666x** [comment]
Looks ok now, thanks.

**Pull request merged**